### ISSUE TITLE
How to order group_options in admin page ?

### ISSUE THREAD

**Issue by crying2812** [Codestar/codestar-framework#432]
<img width="581" alt="screen shot 2016-12-21 at 12 57 51 pm" src="https://cloud.githubusercontent.com/assets/7057776/21378994/32da3a0e-c77d-11e6-8560-65a960041d51.png">

As image, i have a list of item of group_options, is possible when i order them by certain field  (date field) ?

**Comment by choijun** [comment]
It's possible @crying2812 Do you want order them before add new/sorted or before save to database ?

**Comment by crying2812** [comment]
Tks for help me.
@choijun  i want to order them after add new an element to dabase

**Comment by choijun** [comment]
Please trying the solution below.

The first, please add custom attribute for date time field in group, see demo http://prntscr.com/dmnrzx
Then open the file **cs-framworks.js** and replace function $.fn.CSFRAMEWORK_GROUP by

<pre>
$.fn.CSFRAMEWORK_GROUP = function() {
    return this.each(function() {

      var _this           = $(this),
          field_groups    = _this.find('.cs-groups'),
          accordion_group = _this.find('.cs-accordion'),
          clone_group     = _this.find('.cs-group:first').clone();

      if ( accordion_group.length ) {
        accordion_group.accordion({
          header: '.cs-group-title',
          collapsible : true,
          active: false,
          animate: 250,
          heightStyle: 'content',
          icons: {
            'header': 'dashicons dashicons-arrow-right',
            'activeHeader': 'dashicons dashicons-arrow-down'
          },
          beforeActivate: function( event, ui ) {
            $(ui.newPanel).CSFRAMEWORK_DEPENDENCY( 'sub' );
          }
        });
      }

      field_groups.sortable({
        axis: 'y',
        handle: '.cs-group-title',
        helper: 'original',
        cursor: 'move',
        placeholder: 'widget-placeholder',
        start: function( event, ui ) {
          var inside = ui.item.children('.cs-group-content');
          if ( inside.css('display') === 'block' ) {
            inside.hide();
            field_groups.sortable('refreshPositions');
          }
        },
        stop: function( event, ui ) {
          ui.item.children( '.cs-group-title' ).triggerHandler( 'focusout' );
          accordion_group.accordion({ active:false });
        }
      });

      var i = 0;
      $('.cs-add-group', _this).on('click', function( e ) {

        e.preventDefault();
        var c_count = $('.cs-group', field_groups).length;
        clone_group.find('input, select, textarea').each( function () {
          this.name = this.name.replace(/\[(\d+)\]/,function(string, id) {
            return '[' + ( (c_count - 1) +1) + ']';
          });
        });

        var cloned = clone_group.clone().removeClass('hidden');
        field_groups.append(cloned);

        if ( accordion_group.length ) {
          field_groups.accordion('refresh');
          field_groups.accordion({ active: cloned.index() });
        }

        field_groups.find('input, select, textarea').each( function () {
          this.name = this.name.replace('[_nonce]', '');
        });

        field_groups.trigger('la_change_position_field');

        // run all field plugins
        cloned.CSFRAMEWORK_DEPENDENCY( 'sub' );
        cloned.CSFRAMEWORK_RELOAD_PLUGINS();

        i++;

      });

      field_groups.on('click', '.cs-remove-group', function(e) {
        e.preventDefault();
        $(this).closest('.cs-group').remove();
        field_groups.trigger('la_change_position_field');
      });

      field_groups.on('la_change_position_field', function(e){
        // before we need sorted;
        var _this = $(this),
          $group = $('.cs-group', _this);
        $group.sort(function(a,b){
          var an = new Date( $(a).find('[data-alt-class="test"]').val()).getTime(),
              bn = new Date( $(b).find('[data-alt-class="test"]').val()).getTime();

          if(an > bn) {
            return 1;
          }
          if(an < bn) {
            return -1;
          }
          return 0;
        });
        $group.detach().appendTo( _this );

        // then change key of group
        $group.each(function(idx){
          $(this).find('input, select, textarea').each( function () {
            this.name = this.name.replace(/\[(\d+)\]/,function(string, id) {
              return '[' + idx + ']';
            });
          });
        })
      });
    });
  };</pre>

I hope that will be help for you.
Regards,
Jun
